src/utils/CustomerPoolGenerator.java

```diff
@@ -0,0 +1,19 @@
+package utils;
+
+import entity.Customer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Semaphore;
+
+public class CustomerPoolGenerator {
```

Comment:
Generate customers in Runner class. You don't need CustomerPoolGenerator separately.